okapi-core/src/test/java/org/folio/okapi/service/impl/DockerModuleHandleTest.java

```diff
@@ -136,6 +150,110 @@ private void dockerMockHandle(RoutingContext ctx) {
     }
   }
 
+  @Test
+  public void testDockerPull(TestContext context) {
+    Vertx vertx = Vertx.vertx();
+    int dockerPort = 9231;
+
+    Router router = Router.router(vertx);
+    router.routeWithRegex("/.*").handler(this::dockerMockHandle);
+
+    HttpServerOptions so = new HttpServerOptions().setHandle100ContinueAutomatically(true);
+    HttpServer listen = vertx.createHttpServer(so)
+        .requestHandler(router)
+        .listen(dockerPort, context.asyncAssertSuccess());
+    LaunchDescriptor ld = new LaunchDescriptor();
+    ld.setWaitIterations(2);
+    ld.setDockerImage("folioci/mod-x");
+    ld.setDockerPull(true);
+
+    Ports ports = new Ports(9232, 9233);
+    JsonObject conf = new JsonObject().put("dockerUrl", "tcp://localhost:" + dockerPort);
+
+    dockerPullJson = new JsonObject().put("message", "some message");
+    dockerPullStatus = 200;
+
+    {
+      DockerModuleHandle dh = new DockerModuleHandle(vertx, ld,
+          "mod-users-5.0.0-SNAPSHOT", ports, "localhost",
+          9231, conf);
+      Async async = context.async();
+      dh.pullImage().onComplete(context.asyncAssertSuccess(x -> {
+        async.complete();
+      }));
+      async.await();
+    }
+
+    conf.put("dockerRegistries", new JsonArray());
+    {
+      DockerModuleHandle dh = new DockerModuleHandle(vertx, ld,
+          "mod-users-5.0.0-SNAPSHOT", ports, "localhost",
+          9231, conf);
+      Async async = context.async();
+      dh.pullImage().onComplete(context.asyncAssertFailure(x -> {
+        async.complete();
+      }));
+      async.await();
+    }
+
+    conf.put("dockerRegistries", new JsonArray()
+        .addNull()
+        .add(new JsonObject().put("username", "x").put("password", "y")));
```

Comment:
Can you also add a test with an empty JsonObject for unauthenticated pull?
```
conf.put("dockerRegistries", new JsonArray()
        .add(new JsonObject())
        .add(new JsonObject().put("username", "x").put("password", "y")));
```

Reply:
sure

Reply:
I still cannot find such a test that should succeed.

Reply:
I still don't see it, can you point me to the line?

Reply:
conf.put("dockerRegistries", new JsonArray());
    context.assertFalse(pullImage(context, vertx, "localhost", conf));

doesn't succeed. Why should that be a special case?